src/pages/benchmarks/[benchmark_id]/index.tsx

```diff
@@ -0,0 +1,220 @@
+import Counter from "@/components/para_trials/counter";
+import $box from "@/styles/Box.module.css";
+import $button from "@/styles/Button.module.css";
+import { trpc } from "@/client/lib/trpc";
+import { useRouter } from "next/router";
+import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import ParaNav from "@/components/ParaNav";
+import Link from "next/link";
+import { useDebounce } from "react-use";
+
+interface dataUpdate {
+  success_with_prompt?: number;
+  success_without_prompt?: number;
+  submitted?: boolean;
+  notes?: string;
+}
+
+const BenchmarkPage = () => {
+  const router = useRouter();
+  const { benchmark_id } = router.query;
+  const utils = trpc.useContext();
+  const {
+    data: task,
+    isLoading: taskIsLoading,
+    isError,
+  } = trpc.iep.getSubgoalAndTrialData.useQuery(
+    {
+      task_id: benchmark_id as string,
+    },
+    {
+      enabled: Boolean(benchmark_id),
+    }
+  );
+  const seenMutation = trpc.iep.markAsSeen.useMutation({
+    onSuccess: async () => await utils.iep.getSubgoalAndTrialData.invalidate(),
+  });
+  const addTrialMutation = trpc.iep.addTrialData.useMutation({
+    onSuccess: async () => await utils.iep.getSubgoalAndTrialData.invalidate(),
+  });
+  const updateTrialMutation = trpc.iep.updateTrialData.useMutation({
+    onSuccess: async () => await utils.iep.getSubgoalAndTrialData.invalidate(),
+  });
+
+  const [notesValue, setNotesValue] = useState("");
+  // const [runningTotal, setRunningTotal] = useState(0);
+  const [currentTrialIdx, setCurrentTrialIdx] = useState(0);
+  const currentTrial = task?.trials[currentTrialIdx] || null;
+  useEffect(() => {
+    if (task && task.trials.length > 0) {
+      setCurrentTrialIdx(task.trials.length - 1);
+    }
+  }, [task]);
+
+  useEffect(() => {
+    if (task && currentTrial) {
+      // setRunningTotal(
+      //   currentTrial.success_with_prompt + currentTrial.success_without_prompt
+      // );
+      setNotesValue(currentTrial.notes || "");
+    }
+  }, [task, currentTrial]);
+
+  useEffect(() => {
+    if (!seenMutation.isLoading && !taskIsLoading && task && !task.seen) {
+      seenMutation.mutate({ task_id: task.task_id });
+    }
+  }, [task, seenMutation, taskIsLoading]);
+
+  useEffect(() => {
+    if (
+      !addTrialMutation.isLoading &&
+      !taskIsLoading &&
+      task &&
+      (task.trials.length === 0 ||
+        task.trials[task.trials.length - 1]?.submitted === true)
+    ) {
+      addTrialMutation.mutate({
+        task_id: task.task_id,
+        success_with_prompt: 0,
+        success_without_prompt: 0,
+        notes: "",
+      });
+    }
+  }, [task, addTrialMutation, taskIsLoading]);
+
+  const handleUpdate = (updates: dataUpdate) => {
+    //Can only update if we're on the most recent trial
+    if (task && currentTrial && currentTrialIdx === task.trials.length - 1) {
```

Comment:
this logic isn't quite clear to me, why can't we update previous trials?

Reply:
The Paraflow design team stated that previous trials are available for review only and won't be editable.